File: data_binding/src/main/java/me/destro/android/databinding/MainActivity.java
package me.destro.android.databinding;

import android.content.Intent;
import androidx.databinding.DataBindingUtil;
import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import me.destro.android.databinding.basic.BasicActivity;
import me.destro.android.databinding.databinding.ActivityExamplesBinding;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ActivityExamplesBinding binding = DataBindingUtil.setContentView(this, R.layout.activity_examples);


        binding.basic.getRoot().setOnClickListener(view -> {
            Intent intent = new Intent(this, BasicActivity.class);
            startActivity(intent);
        });

        binding.basic.title.setText(R.string.basic_example_title);
        binding.basic.description.setText(R.string.basic_example_description);
    }
}

File: navigation/src/main/java/me/destro/android/navigation/MainActivity.kt
package me.destro.android.navigation

import android.content.Intent
import androidx.databinding.DataBindingUtil
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import me.destro.android.navigation.databinding.ActivityExamplesBinding
import me.destro.android.navigation.fullexample.FullExampleActivity
import me.destro.android.navigation.databinding.ActivityNavigationHostBinding

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityNavigationHostBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val binding = DataBindingUtil.setContentView<ActivityExamplesBinding>(this, R.layout.activity_examples)

        binding.basic.root.setOnClickListener { view ->
            val intent = Intent(this, FullExampleActivity::class.java)
            startActivity(intent)
        }

        binding.basic.title.text = "Full example"
        binding.basic.description.text = "Description"
    }

}
